docs/src/references/agent/device-management-api.md

```diff
@@ -0,0 +1,330 @@
+
+```sh te2mqtt formats="v1"
+tedge mqtt pub --retain 'te/device/child001///cmd/software-update' '{
+    "description": "Install, update and remove software packages",
+    "supportedTypes": [
+      "apt",
+      "docker",
+    ]
+}'
+```
+
+On reception of this message, the Cumulocity mapper notifies Cumulocity of this capability.
+On request from a cloud operator, the Cumulocity mapper creates a new command instance, say to update `nodered`:
+
+```sh te2mqtt formats="v1"
+tedge mqtt pub --retain 'te/device/child001///cmd/software-update/c8y-123' '{
+    "status": "init",
+    "updateList": [
+        {
+            "type": "debian",
+            "modules": [
+                {
+                    "name": "nodered",
+                    "version": "1.0.0",
+                    "action": "install"
+                }
+            ]
+        }
+    ]
+}'
+```
+
+The agent, running on `device/child001//`, notifies that it will execute the command:
+
+```sh te2mqtt formats="v1"
+tedge mqtt pub --retain 'te/device/child001///cmd/software-update/c8y-123' '{
+    "status": "executing",
+    "updateList": [
+        {
+            "type": "debian",
+            "modules": [
+                {
+                    "name": "nodered",
+                    "version": "1.0.0",
+                    "action": "install"
+                }
+            ]
+        }
+    ]
+}'
+```
+
+Then the agent proceeds, here installing a specific version of `nodered`, and notifies the mapper when done:
+
+```sh te2mqtt formats="v1"
+tedge mqtt pub --retain 'te/device/child001///cmd/software-update/c8y-123' '{
+    "status": "successful",
+    "updateList": [
+        {
+            "type": "debian",
+            "modules": [
+                {
+                    "name": "nodered",
+                    "version": "1.0.0",
+                    "action": "install"
+                }
+            ]
+        }
+    ],
+    "currentSoftwareList": [
+        {
+            "type": "debian",
+            "modules": [
+                {
+                    "name": "nodered",
+                    "version": "1.0.0",
+                },
+                {
+                    "name": "collectd",
+                    "version": "5.7"
+                }
+            ]
+        },
+        {
+            "type": "docker",
+            "modules": [
+                {
+                    "name": "nginx",
+                    "version": "1.21.0",
+                },
+                {
+                    "name": "mongodb",
+                    "version": "4.4.6",
+                }
+            ]
+        }
+    ]
+}'
+```
+
+The Cumulocity mapper, having subscribed to all software-update commands,
+monitors this command instance and notifies Cumulocity of its progress upto completion.  
+Finally, the Cumulocity mapper clear the command topic:
+
+```sh te2mqtt formats="v1"
+tedge mqtt pub --retain 'te/device/child001///cmd/software-update/c8y-123' ''
+```
```

Comment:
I'm assuming that the workflow API bits like the workflow definition file from [this PR](https://github.com/thin-edge/thin-edge.io/pull/2071) will also be added to the device-mangement-api doc at some point. Because, that'll give clarity to the users on how different components can work together with `tedge-agent` to drive an operation to completion.

Reply:
Yes, definitely. I even think the whole [operation workflow specification](https://github.com/thin-edge/thin-edge.io/pull/2071) has to reworked on top of this PR.